test-case-reporting/test/test_result_record.test.ts

```diff
@@ -0,0 +1,301 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {DB, KarmaReporter, Travis} from 'test-case-reporting';
+import {Database, dbConnect} from '../src/db';
+import {TestResultRecord} from '../src/test_result_record';
+import {getFixture} from './fixtures';
+import {setupDb} from '../src/setup_db';
+import {truncateAll} from './testing_utils';
+import Knex from 'knex';
+
+jest.mock('../src/db', () => ({
+  dbConnect: (): Database =>
+    Knex({
+      client: 'sqlite3',
+      connection: ':memory:',
+      useNullAsDefault: true,
+    }),
+}));
+
+describe('TestResultRecord', () => {
+  let db: Database;
+  let testResultRecord: TestResultRecord;
+
+  const sampleBuild: Travis.Build = {
+    buildNumber: '413413',
+    commitSha: 'abcdefg123gomugomu',
+  };
+
+  const sampleJob: Travis.Job = {
+    jobNumber: '413413.612',
+    testSuiteType: 'unit',
+  };
+
+  const sampleKarmaReport: KarmaReporter.TestResultReport = (getFixture(
+    'sample-karma-report'
+  ) as unknown) as KarmaReporter.TestResultReport;
+
+  const sampleTravisReport: Travis.Report = {
+    job: sampleJob,
+    build: sampleBuild,
+    result: sampleKarmaReport,
+  };
+
+  beforeAll(async () => {
+    db = dbConnect();
+    await setupDb(db);
+
+    testResultRecord = new TestResultRecord(db);
```

Comment:
Since this is the object under tests, it's usually a good idea to create a new one for each test case, so let's put this assignment into a `beforeEach` instead. the rest looks good(

Reply:
Done.